### DISCUSSION TITLE
Bug in validation dataset name

### DISCUSSION THREAD

**Discussion by donhuvy** [ultralytics/ultralytics#24682] [Q&A] [answered]
Bug in validation dataset name.

Dear team,

At this link https://platform.ultralytics.com/ultralytics/yolo26 , I think this is a bug. I tried 10 different file names for the datasets, but I was unsuccessful.

Please see the attached file.

Best regards,

<img width="1910" height="995" alt="dset" src="https://github.com/user-attachments/assets/4c64ffe0-c578-4265-9fc9-f8eebe314563" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @donhuvy, thank you for reporting this potential bug in the validation dataset name workflow on the Ultralytics Platform 🚀 This is an automated response to help gather the right details quickly, and an Ultralytics engineer will also assist you soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. 🙏
Useful details include:
- The exact dataset names you tried
- Whether the issue happens for training datasets, validation datasets, or both
- The file type and structure of the uploaded dataset
- Any error messages shown in the UI
- A short step-by-step sequence to reproduce the problem consistently
- Additional screenshots if available

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
If this is a dataset YAML issue rather than a Platform UI bug, the validation split must be defined as `val:` in `data.yaml` alongside `train:`; `validation:` is auto-mapped, but `valid:` or custom split keys are not, per the [dataset YAML checks](https://docs.ultralytics.com/reference/data/utils#ultralytics.data.utils.check_det_dataset), so please paste your exact `data.yaml` and the UI error text here as text and we’ll check it.

**Comment by donhuvy** [comment]
Thank you, bro. I know you YOLO's founder 👍 . I tried to upload my dataset (object detection), Is 5 GB and 2000 images (30200 images objects) ok for uploading? Maybe you consider make user interace more easier to understand. I did not know what is next step . I train model at local success, just try to upload online for impression my master thesis supervisor.

My data set

<img width="1199" height="1919" alt="image" src="https://github.com/user-attachments/assets/46b2a6bc-87c6-475b-a7b1-c5bb26b99a44" />

for this mobile app https://score.donhuvy.com/

**Reply by banu4prasad** [reply]
> Is 5 GB and 2000 images (30200 images objects) ok for uploading?

Ig the upload limit is 10 GB on platform.ultralytics.com

> just try to upload online for impression my master thesis supervisor.

For that create a folder like <multi_choice....> and upload ur trained model here first then u can deploy later

**Reply by glenn-jocher** [reply]
Yes, `5 GB` and `2000` images are within the supported Platform limits as long as your dataset is packaged in a valid YOLO structure with a `data.yaml` plus `train` and `val` splits; after upload finishes processing, open the dataset and click `New Model` to start training, and the [Platform dataset guide](https://docs.ultralytics.com/platform/data/datasets) and [quickstart](https://docs.ultralytics.com/platform/quickstart) show the expected layout and next steps.

**Comment by donhuvy** [comment]
<img width="1199" height="1919" alt="image" src="https://github.com/user-attachments/assets/0692ba12-7e41-46ae-b0bd-078a4f8128a8" />

<img width="1199" height="1919" alt="image" src="https://github.com/user-attachments/assets/b9d764dc-72cc-4eb4-a647-b402806c1331" />

**Comment by donhuvy** [comment]
I have file `C:\Users\ADMIN\PyCharmMiscProject\brain-tumor.yaml` has full content:

```yaml
# Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license

# Brain-tumor dataset by Ultralytics
# Documentation: https://docs.ultralytics.com/datasets/detect/brain-tumor/
# Example usage: yolo train data=brain-tumor.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── brain-tumor ← downloads here (4.21 MB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: brain-tumor # dataset root dir
train: images/train # train images (relative to 'path') 893 images
val: images/val # val images (relative to 'path') 223 images
test: # test images (relative to 'path')

# Classes
names:
  0: A
  1: B
  2: C
  3: D

# Download script/URL (optional)
download: https://github.com/ultralytics/assets/releases/download/v0.0.0/brain-tumor.zip
```

**Reply by glenn-jocher** [reply, marked as answer]
The YAML filename is not the issue here; per the [Brain Tumor dataset docs](https://docs.ultralytics.com/datasets/detect/brain-tumor), the official `brain-tumor.yaml` is a 2-class dataset (`negative`, `positive`) with `train: images/train` and `val: images/val`, so your `names` block with 4 classes will only work if your label files also use class IDs `0-3`. After correcting the YAML/labels, delete any `labels/*.cache` files and retry, and if it still fails please paste the full text error plus `yolo checks` output.
